### ISSUE TITLE
regarding pull request - https://github.com/Swiip/generator-gulp-angular/pull/545

### ISSUE THREAD

**Issue by amirtuval** [Swiip/generator-gulp-angular#597]
I added a comment in this pull request, but then I noticed it was rejected and closed, so I am opening an issue here, just in case my comment goes unnoticed.
The pull requests fixes an issue with the proxy mechanism, which to my understanding, does not work currently .

Here's what I wrote - 

I encountered this too.

I'll explain. All files under the gulp directory should export a function, otherwise you'll get an error.
However, server.js expects proxy.js to export an array of middleware, or the proxy.js setting will not work.

The above fixes this by not requiring proxy.js thus avoiding the error.

Another option is to change server.js to expect proxy.js to export a function.

Basically change

    var middleware = require('./proxy');

to

    var middleware = require('./proxy')();

**Comment by Swiip** [comment]
This is quite obsolete. In master, gulp directory modules doesn't export functions anymore.

In addition, we closed the PR because we just switched on an external lib for the proxy https://www.npmjs.com/package/http-proxy-middleware and we don't need any `require('./proxy')` anymore.

**Comment by zckrs** [comment]
:tada: Just released v0.12.0 :tada:

Your issue should be fixed
